### PR TITLE
Merge paradox/scala/howto.md and java/howto.md 

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #23052

## Rendered HTML:

- https://richard-akka-doc.firebaseapp.com/scala/howto.html
- https://richard-akka-doc.firebaseapp.com/java/howto.html

## I split this in two commits:

- **1st commit:** Identical contents in scala/howto.md and java/howto.md -> **this is to take the diff both in scala/howto.md and java/howto.md**

  - [scala/howto.md  (prev) vs. scala/howto.md (after)](https://github.com/richard-imaoka/akka/commit/a6a5628c153cc717c2c1ae8a4d2649ff51daad8d#diff-91687dc400453b9934536c8baafb3ac3)
  - [java/howto.md (prev) vs. java/howto.md (after)](https://github.com/richard-imaoka/akka/commit/a6a5628c153cc717c2c1ae8a4d2649ff51daad8d#diff-d5a161c63bdea4a5a6e0d66411c53c40)

- **2nd commit:** Replace java/howto.md with a symlink to scala/howto.md

### CODE REVIEW

File: akka-docs/src/main/paradox/scala/howto.md [hunk 1]
```diff
@@ -8,6 +8,15 @@ would be nice if every Akka user who finds a recurring pattern in his or her
...
 
+@@@ div { .group-java }
+
+You might find some of the patterns described in the Scala chapter of 
+@ref:[HowTo: Common Patterns](../scala/howto.md) useful even though the example code is written in Scala.
+
+@@@ 
+
+@@@ div { .group-scala }
+
```
Comment (richard-imaoka) [inline akka-docs/src/main/paradox/scala/howto.md]: From here, Scala patterns are listed.

File: akka-docs/src/main/paradox/scala/howto.md [hunk 2]
```diff
@@ -141,4 +156,49 @@ under pressure, but only schedule a new tick message when we have seen the previ
...
-@@snip [SchedulerPatternSpec.scala]($code$/scala/docs/pattern/SchedulerPatternSpec.scala) { #schedule-receive }
\ No newline at end of file
+Scala
+:  @@snip [SchedulerPatternSpec.scala]($code$/scala/docs/pattern/SchedulerPatternSpec.scala) { #schedule-receive }
+
+Java
+:  @@snip [SchedulerPatternTest.java]($code$/java/jdocs/pattern/SchedulerPatternTest.java) { #schedule-receive }
+
+@@@ div { .group-java }
+
```
Comment (richard-imaoka) [inline akka-docs/src/main/paradox/scala/howto.md]: The below pattern is specific to Java.

> A nice way to enter the actor world from java is the use of Patterns.ask().

File: akka-docs/src/main/paradox/java/howto.md [hunk 3]
```diff
@@ -8,8 +8,117 @@ would be nice if every Akka user who finds a recurring pattern in his or her
 code could share it for the profit of all. Where applicable it might also make
 sense to add to the `akka.pattern` package for creating an [OTP-like library](http://www.erlang.org/doc/man_index.html).
 
+@@@ div { .group-java }
+
 You might find some of the patterns described in the Scala chapter of 
-@ref:[HowTo: Common Patterns](howto.md) useful even though the example code is written in Scala.
+@ref:[HowTo: Common Patterns](../scala/howto.md) useful even though the example code is written in Scala.
```
Comment (raboof) [inline akka-docs/src/main/paradox/java/howto.md]: that's the page you're on right now, right? so I'd replace `@ref:[HowTo: Common Patterns](howto.md)` with simply "this page".

**Review by raboof — CHANGES_REQUESTED** [review]
Tiny nitpick otherwise looks good!

File: akka-docs/src/main/paradox/java/howto.md (hunk 3 above)
Comment (richard-imaoka) [inline akka-docs/src/main/paradox/java/howto.md]: The line is enclosed in a `@@@ div { .group-java }` block.
image

The reference is intended to jump from `java/howto.md` to `scala/howto.md`. 

Instead, if we replace `(../scala/howto.md)` with `(howto.md)`, then you jump from` java/howto.md` to `java/howto.md`, the same page, which is not correct.

Comment (raboof) [inline akka-docs/src/main/paradox/java/howto.md]: I meant with the plain text "this page", not a reference at all. People can switch languages using the dropdown in the top-left of the screen, after all.

Once all pages have been converted we can drop the `scala`/`java` prefix, and `../scala` will no longer exist.

Comment (richard-imaoka) [inline akka-docs/src/main/paradox/java/howto.md]: Ah I see. Sorry I misunderstood your first comment.

Fixed it in the new commit.

**Pull request merged**